README.md

```diff
@@ -0,0 +1,54 @@
+PR Watcher Notifier
+===================
+
+A flask web application that handles webhook notifications for pull request changes and sends out notifications
+when there are changes to files matched by the configured patterns for each repository
+
+
+Pre-requisites
+==============
+
+Python 3.6 or newer
+
+
+GitHub Webhook setup
+====================
+
+* Use a tool like `ngrok` to expose the local development server on the internet.
+* Create a webhook on the repository to be watched by following [these instructions](https://developer.github.com/webhooks/creating/).
+This requires administrator-level access to the repository.
+* Use the `<ngrok URL>/pull-requests` as the payload URL and set the `Content Type` to `application/json`.
+* Remember to set a secret and to select only the 'Pull request' event.
+* Create a GitHub web token.
+
+
+Development Setup
+=================
+
+* Clone this repository and navigate into the cloned directory.
+* Create a virtualenv environment and install the requirements.
+
+```bash
+$ python3 -m venv venv
+$ source venv/bin/activate
+$ pip install -r requirements.txt -r requirements-dev.txt
+```
+
+* Verify that the unit tests run without errors and pass.
+
+```bash
+$ pytest
+```
+
+* Configure the environment variables used in the `settings.py` file.
+* Configure the secrets, repositories to watch, the file patterns, the email subject and the recipients in the same file.
+* Set the `FLASK_APP` environment variable to point to the `run.py` file.
+* Set the `FLASK_DEBUG` environment variable to `1` and run the development server.
+
+```bash
+$ export FLASK_APP=run.py
+$ export FLASK_DEBUG=1
+$ flask run
```

Comment:
To use the app on my non-localhost domain, I had to use `flask run --host=0.0.0.0`.  Is there any way to make that IP configurable from the settings or env?

Reply:
@pomegranited, the `flask run` command doesn't read from settings or the environment variables. The old way of writing a file with `app.run(host='0.0.0.0')` in the `if __name__ == '__main__'` block of a script file is not recommended any more.

If there is time, I can create a script `run_dev.sh` reads the environment variable if it is set and modifies the command appropriately. Otherwise, we might have to stop with documenting this.

Reply:
👍